resources/conf/wdio-bstack-local-parallel-devices.conf.js

```diff
@@ -0,0 +1,101 @@
+var defaults = require("./wdio.conf.js");
+var _ = require("lodash");
+
+var overrides = {
+  onBrowserstack: true,
+
+  user: process.env.BROWSERSTACK_USER,
+  key: process.env.BROWSERSTACK_ACCESSKEY,
+
+  specs: ["./test/specs/local/local.spec.js"],
+
+  capabilities: [
+    {
+      maxInstances: 1,
+      name:
+        require("minimist")(process.argv.slice(2))["bstack-session-name"] ||
+        "default_name",
+      build:
+        process.env.BROWSERSTACK_BUILD_NAME ||
+        "browserstack-examples-appium-webdriverio" +
+          " - " +
+          new Date().getTime(),
+      project: "browserstack-examples-appium-webdriverio",
+      device: "Google Pixel 5",
+      os_version: "11.0",
+      app: process.env.BROWSERSTACK_ANDROID_APP_ID,
+      platformName: "Android",
+      "browserstack.local": true,
+      autoGrantPermissions: true,
+    },
+    {
+      maxInstances: 1,
+      name:
+        require("minimist")(process.argv.slice(2))["bstack-session-name"] ||
+        "default_name",
+      build:
+        process.env.BROWSERSTACK_BUILD_NAME ||
+        "browserstack-examples-appium-webdriverio" +
+          " - " +
+          new Date().getTime(),
+      project: "browserstack-examples-appium-webdriverio",
+      device: "iPhone 12 Pro",
+      os_version: "14",
+      app: process.env.BROWSERSTACK_IOS_APP_ID,
+      platformName: "iOS",
+      automationName: "XCUITest",
+      "browserstack.local": true,
+      gpsEnabled: "true",
+    },
+  ],
+
+  logLevel: "warn",
+  bail: 0,
+  baseUrl: "http://localhost",
+  waitforTimeout: 10000,
+  connectionRetryTimeout: 240000,
+  connectionRetryCount: 3,
+  services: ["browserstack"],
+  framework: "mocha",
+  reporters: ["spec", ["allure", { outputDir: "allure-results" }]],
+  mochaOpts: {
+    ui: "bdd",
+    timeout: 240000,
+  },
+  afterTest: async function (
+    test,
+    context,
+    { error, result, duration, passed, retries }
+  ) {
+    if (require("minimist")(process.argv.slice(2))["bstack-session-name"]) {
+      await driver.executeScript(
+        'browserstack_executor: {"action": "setSessionName", "arguments": {"name":"' +
+          require("minimist")(process.argv.slice(2))["bstack-session-name"] +
+          '" }}',
+        []
+      );
+    } else {
+      await driver.executeScript(
+        'browserstack_executor: {"action": "setSessionName", "arguments": {"name":"' +
+          test.title +
+          '" }}',
+        []
+      );
+    }
+
+    if (passed) {
+      await driver.executeScript(
+        'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "Assertions passed"}}',
+        []
+      );
+    } else {
+      await driver.takeScreenshot();
+      await browser.executeScript(
+        'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "At least 1 assertion failed"}}',
+        []
+      );
+    }
+  },
```

Comment:
this segment (test hook) is getting repeated in a number of files.

we could create this file extends wdio-bstack.conf.js extends wdio.conf.js.

and then have bstack common hooks in wdio-bstack.conf.js. 3 level hierarchy.

Reply:
`wdio.conf.js` will be extended by `wdio-bstack.conf.js` which will in turn be extended by individual confs.